include/simple-mpc/interpolator.hpp

```diff
@@ -0,0 +1,53 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2025, INRIA
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+/**
+ * @file interpolator.hpp
+ * @brief Interpolation class for practical control of the robot
+ */
+
+#ifndef SIMPLE_MPC_INTERPOLATOR_HPP_
+#define SIMPLE_MPC_INTERPOLATOR_HPP_
+
+#include "simple-mpc/fwd.hpp"
+
+namespace simple_mpc
+{
+  class Interpolator
+  {
+  public:
+    explicit Interpolator(const long nx, const long nv, const long nu, const long nf, const double MPC_timestep);
+
+    void interpolate(
+      const double delay,
+      std::vector<Eigen::VectorXd> xs,
+      std::vector<Eigen::VectorXd> us,
+      std::vector<Eigen::VectorXd> ddqs,
+      std::vector<Eigen::VectorXd> forces);
```

Comment:
To be discussed, but maybe we could have 2 interpolate method, one for the configuration vector (that will rely on the exp/log) and one for the tangent space, or any other quantities (that can just be linearly interpolated).

Because the user might want to interpolate, or not, each quantity (q, dq, ddq, tau, forces) or more.
Forcing to pass exactly all of them can be an unnecessary constraint.